tools/tpm2_getmanufec.c

```diff
@@ -184,36 +184,35 @@ int createEKHandle(TSS2_SYS_CONTEXT *sapi_context)
     /*
      * use enAuth in Tss2_Sys_CreatePrimary
      */
-    if (strlen(ctx.endorse_passwd) > 0 && !ctx.hex_passwd) {
-            sessionData.hmac.t.size = strlen(ctx.endorse_passwd);
-            memcpy( &sessionData.hmac.t.buffer[0], ctx.endorse_passwd, sessionData.hmac.t.size );
-    }
-    else {
-        if (strlen(ctx.endorse_passwd) > 0 && ctx.hex_passwd) {
-                sessionData.hmac.t.size = sizeof(sessionData.hmac) - 2;
-
-                if (tpm2_util_hex_to_byte_structure(ctx.endorse_passwd, &sessionData.hmac.t.size,
-                                      sessionData.hmac.t.buffer) != 0) {
-                        LOG_ERR("Failed to convert Hex format password for endorsePasswd.");
-                        return 1;
-                }
+    if (ctx.endorse_passwd && strlen(ctx.endorse_passwd) > 0) {
+        if (!ctx.hex_passwd) {
+            memcpy(&sessionData.hmac.t.buffer[0], ctx.endorse_passwd,
+                   sessionData.hmac.t.size );
+        } else {
+            sessionData.hmac.t.size = sizeof(sessionData.hmac) - 2;
+
+            if (tpm2_util_hex_to_byte_structure(ctx.endorse_passwd,
```

Comment:
Oooh old -X option code...

Reply:
Update this to use the password util routines, that's my only request for this PR, the rest looks great.